pygame_gui/elements/ui_button.py

```diff
@@ -335,53 +348,59 @@ def process_event(self, event: pygame.event.Event) -> bool:
                 self._set_inactive()
                 consumed_event = True
                 self.pressed_event = True
-                self.on_pressed(event.button)
+                self.on_event(UI_BUTTON_PRESSED, {'mouse_button':event.button})
 
             if self.is_enabled and self.held:
                 self.held = False
                 self._set_inactive()
                 consumed_event = True
 
         return consumed_event
+    
+    def bind(self, event:int, function:Callable = None):
+        """
+        Bind a function to an event.
 
-    def on_start_press(self, button: int):
-        # old event to remove in 0.8.0
-        event_data = {'user_type': OldType(UI_BUTTON_START_PRESS),
-                        'ui_element': self,
-                        'ui_object_id': self.most_specific_combined_id}
-        pygame.event.post(pygame.event.Event(pygame.USEREVENT, event_data))
+        :param event: The event to bind.
 
-        # new event
-        event_data = {'ui_element': self,
-                        'ui_object_id': self.most_specific_combined_id,
-                        'mouse_button': button}
-        pygame.event.post(pygame.event.Event(UI_BUTTON_START_PRESS, event_data))
-
-    def on_pressed(self, button: int):
-        # old event
-        event_data = {'user_type': OldType(UI_BUTTON_PRESSED),
-                        'ui_element': self,
-                        'ui_object_id': self.most_specific_combined_id}
-        pygame.event.post(pygame.event.Event(pygame.USEREVENT, event_data))
+        :param function: The function to bind. None to unbind.
 
-        # new event
-        event_data = {'ui_element': self,
-                        'ui_object_id': self.most_specific_combined_id,
-                        'mouse_button': button}
-        pygame.event.post(pygame.event.Event(UI_BUTTON_PRESSED, event_data))
+        """
+        if function is None:
+            self.handler.pop(event, None)
+            return
 
-    def on_double_clicked(self, button: int):
-        # old event to remove in 0.8.0
-        event_data = {'user_type': OldType(UI_BUTTON_DOUBLE_CLICKED),
-                        'ui_element': self,
-                        'ui_object_id': self.most_specific_combined_id}
-        pygame.event.post(pygame.event.Event(pygame.USEREVENT, event_data))
+        if callable(function):
+            self.handler[event] = function
+        else:
+            raise TypeError("Function to bind must be callable")
+    
+    def on_event(self, event:int, data:Dict[str, Any]=None):
```

Comment:
```suggestion
    def on_click_event(self, event:int, data:Dict[str, Any]=None):
```
Since this is all about click events.